code/jobtap-plugins/state-counts.c

```diff
@@ -0,0 +1,132 @@
+/************************************************************\
+ * Copyright 2014 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#include <flux/core.h>
+#include <flux/jobtap.h>
+
+#include "../fripp.h"
+
+struct state_counts {
+    int new;
+    int depend;
+    int priority;
+    int sched;
+    int run;
+    int cleanup;
+    int inactive;
+
+    struct fripp_ctx *fripp;
+    flux_watcher_t *w;
+    flux_plugin_t *p;
+};
+
+static void count (struct state_counts *c, flux_job_state_t state, int inc)
+{
+    switch (state) {
+        case FLUX_JOB_STATE_NEW:
+            c->new += inc;
+            break;
```

Comment:
You might want to omit any action for the NEW state, since as you indicated in your journal, it will appear in the "last" state but never the current state in the state callback, so the count will run negative.

Reply:
I think NEW may be the "current state" in the `job.new` callback, but one issue is that the `job.new` callback does not set a `prev_state`. It may indeed just be easier to ignore `FLUX_JOB_STATE_NEW`

Reply:
Oops, sorry should have checked code before commenting.  Thanks.